test/unit_tests/data_structures/array_test.cc

```diff
@@ -367,3 +367,57 @@ TEST_F(ArrayTest, AsyncInsertAsyncForEachAndAsyncGet) {
   }
   shad::Array<size_t>::Destroy(edsPtr->GetGlobalID());
 }
+
```

Comment:
It may be better to move constant values (e.g. the 1234 sizes) from the individual tests, and use const params instead.

Reply:
I know it doesn't look good, but the whole test is designed to make iterators jump back and forth across the boundary of a locality.

I will try to implement that with some costexpr variable so that the test will stimulate the same case changing array size and number of localities.